ChatQnA/Dockerfile

```diff
@@ -1,31 +1,51 @@
-
-
 # Copyright (C) 2024 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-FROM python:3.11-slim
+ARG IMAGE_NAME=python
+ARG IMAGE_TAG=3.11-slim
+FROM ${IMAGE_NAME}:${IMAGE_TAG} as base
 
-RUN apt-get update -y && apt-get install -y --no-install-recommends --fix-missing \
+ENV LANG C.UTF-8
+
+RUN apt-get update -y && apt-get install -y \
     libgl1-mesa-glx \
-    libjemalloc-dev \
-    git
+    libjemalloc-dev && \
```

Comment:
Why these are installed?   Nothing should be using Mesa / 3D libs, and jemalloc usage would need explicit e.g. LD_PRELOAD for it, which I do not see here.

If those are dependencies of some pip package, pip should take care it, shouldn't it?

Reply:
No unfortunately Pip won't take care of development packages.